src/humancellatlas/data/metadata/helpers/dss.py

```diff
@@ -0,0 +1,79 @@
+from urllib3 import Timeout
+
+from humancellatlas.data.metadata import JSON
+
+logger = logging.getLogger(__name__)
+
+
+def download_bundle_metadata(client: DSSClient,
+                             replica: str,
+                             uuid: str,
+                             version: Optional[str] = None,
+                             num_workers: Optional[int] = None) -> Tuple[str, List[JSON], JSON]:
+    """
+    Download the metadata for a given bundle from the HCA data store (DSS).
+
+    :param client: A DSS API client instance
+
+    :param replica: The name of the DSS replica to use
+
+    :param uuid: The UUID of the bundle in DSS
+
+    :param version: The version of the bundle. if None, the most recent version of the bundle will be downloaded.
+
+    :param num_workers: the size of the thread pool to use for downloading metadata files in parallel. If None, the
+                        default pool size will be used, typically a small multiple of the number of cores on the system
+                        executing this function. If 0, no thread pool will be used and all files will be downloaded
+                        sequentially by the current thread.
+
+    :return: A tuple consisting of the version of the downloaded bundle, a list of the manifest entries for all files
+             in the bundle (data and metadata) and a dictionary mapping the file name of each metadata file in the
+             bundle to the JSON contents of that file.
+    """
+    logger.debug("Getting bundle %s.%s from DSS.", uuid, version)
+    # noinspection PyUnresolvedReferences
+    response = client.get_bundle(uuid=uuid, version=version, replica=replica)
+    bundle = response['bundle']
+    manifest = bundle['files']
+    metadata_files = {f["name"]: f for f in manifest if f["indexed"]}
+
+    def download_file(item):
+        file_name, manifest_entry = item
+        file_uuid = manifest_entry['uuid']
+        file_version = manifest_entry['version']
+        logger.debug("Getting file '%s' (%s.%s) from DSS.", file_name, file_uuid, file_version)
+        # noinspection PyUnresolvedReferences
+        return file_name, client.get_file(uuid=file_uuid, version=file_version, replica='aws')
+
+    if num_workers == 0:
+        metadata_files = map(download_file, metadata_files.items())
+    else:
+        with ThreadPoolExecutor(num_workers) as tpe:
+            metadata_files = tpe.map(download_file, metadata_files.items())
+
+    return bundle['version'], manifest, dict(metadata_files)
+
+
+def dss_client(deployment: Optional[str] = None) -> DSSClient:
+    """
+    Return a DSS client to DSS production or the specified DSS deployment.
+
+    :param deployment: The name of a DSS deployment like `dev`, `integration` or `staging`. If None, the production
+                       deployment (`prod`) will be used.
+    """
+    # Work around https://github.com/HumanCellAtlas/dcp-cli/issues/142
+    hca_config = HCAConfig("hca")
```

Comment:
This raises:
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-9-029257d92355> in <module>()
      1 from humancellatlas.data.metadata.helpers.dss import download_bundle_metadata, dss_client
----> 2 client = dss_client()
      3 uuid = "bd0686b4d-a4b7-45e2-89b7-ac0ebdb4a6b7"
      4 version, manifest, metadata_files = download_bundle_metadata(client, 'aws', uuid)
      5 manifest[0]

~/SDK/Python3Env/lib/python3.6/site-packages/humancellatlas/data/metadata/helpers/dss.py in dss_client(deployment)
     69     """
     70     # Work around https://github.com/HumanCellAtlas/dcp-cli/issues/142
---> 71     hca_config = HCAConfig("hca")
     72     deployment = deployment + "." if deployment else ""
     73     hca_config['DSSClient'].swagger_url = f'https://dss.{deployment}data.humancellatlas.org/v1/swagger.json'

~/SDK/Python3Env/lib/python3.6/site-packages/hca/config.py in __init__(self, *args, **kwargs)
      7 
      8     def __init__(self, *args, **kwargs):
----> 9         super(HCAConfig, self).__init__(name="hca", *args, **kwargs)
     10 
     11     @property

TypeError: __init__() got multiple values for argument 'name'
```
with the example `simple.ipynb`

FYI:
```
hca==4.1.4
hca-metadata-api==1.0a4.dev1
```

Reply:
Thank you. I'll look into it.

Reply:
hca-metadata-api is still on hca==4.1.0 and I am not able to reproduce this with 4.1.0. I will upgrade to 4.1.4 in another PR and resolve this problem there.

Reply:
@hannes-ucsc Sure! Actually as I'm onboarding to this new API lib, simply removing the argument `"hca"` (so line https://github.com/HumanCellAtlas/metadata-api/blob/65f803c128194d59aa5db54183ec93d93ee3354d/src/humancellatlas/data/metadata/helpers/dss.py#L71 becomes `hca_config = HCAConfig()`) solved my problem.